kv/notification_endpoint.go

```diff
@@ -0,0 +1,544 @@
+package kv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/influxdata/influxdb/kit/tracing"
+	"github.com/influxdata/influxdb/notification/endpoint"
+
+	"github.com/influxdata/influxdb"
+)
+
+var (
+	notificationEndpointBucket = []byte("notificationEndpointv1")
+	notificationEndpointIndex  = []byte("notificationEndpointIndexv1")
+
+	// ErrNotificationEndpointNotFound is used when the notification endpoint is not found.
+	ErrNotificationEndpointNotFound = &influxdb.Error{
+		Msg:  "notification endpoint not found",
+		Code: influxdb.ENotFound,
+	}
+
+	// ErrInvalidNotificationEndpointID is used when the service was provided
+	// an invalid ID format.
+	ErrInvalidNotificationEndpointID = &influxdb.Error{
+		Code: influxdb.EInvalid,
+		Msg:  "provided notification endpoint ID has invalid format",
+	}
+)
+
+var _ influxdb.NotificationEndpointService = (*Service)(nil)
+
+func (s *Service) initializeNotificationEndpoint(ctx context.Context, tx Tx) error {
+	if _, err := s.notificationEndpointBucket(tx); err != nil {
+		return err
+	}
+	if _, err := s.notificationEndpointIndexBucket(tx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UnavailableNotificationEndpointStoreError is used if we aren't able to interact with the
+// store, it means the store is not available at the moment (e.g. network).
+func UnavailableNotificationEndpointStoreError(err error) *influxdb.Error {
+	return &influxdb.Error{
+		Code: influxdb.EInternal,
+		Msg:  fmt.Sprintf("Unable to connect to notification endpoint store service. Please try again; Err: %v", err),
+		Op:   "kv/notificationEndpoint",
+	}
+}
+
+// UnavailableNotificationEndpointIndexError is used when the error comes from an internal system.
+func UnavailableNotificationEndpointIndexError(err error) *influxdb.Error {
+	return &influxdb.Error{
+		Code: influxdb.EInternal,
+		Msg:  fmt.Sprintf("unexpected error retrieving notfication endpoint's index bucket; Err %v", err),
+		Op:   "kv/notificationEndpointIndex",
+	}
+}
+
+// InternalNotificationEndpointStoreError is used when the error comes from an
+// internal system.
+func InternalNotificationEndpointStoreError(err error) *influxdb.Error {
+	return &influxdb.Error{
+		Code: influxdb.EInternal,
+		Msg:  fmt.Sprintf("Unknown internal notificationEndpoint data error; Err: %v", err),
```

Comment:
maybe `notificationEndpoint` -> `notification endpoint` for consistency in our messages?